Bottom: f23e40b5d3da33bfab9d2313951aaea0c699ce0e
Top:    16dab66b09154887f51241993ddb7d3e661b4bd4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-24 03:12:23 +0000

Workaround crash where contour is passed as NULL to poly_PreContour








---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
